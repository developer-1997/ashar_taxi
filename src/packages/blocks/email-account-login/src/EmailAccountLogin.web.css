#rightBoxContainer {
    height: auto !important;
    margin: 0;
    padding: 30px;
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: center;
    box-sizing: border-box;
}
.MuiGrid-item{
    width: 100%;
}
#linkToSignUpText {
    font-size: 20px;    
    padding-bottom: 5px;
}
.backToHomeButton {
    color: #828282;
    font-weight: 600;
    font-size: 18px !important;
    text-decoration: underline !important;
    cursor: pointer;
    padding: 0px !important;
}
.contentForm {
    display: flex;
    flex-direction: column;
    justify-content: center;
    box-sizing: border-box;
    min-height: min-content;
    padding-right: 60px;
}
.fullGrid {
    margin: 4rem 0 0 !important;
}
.logoContainerStyle { 
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    width: 100%;
    margin-bottom: 40px;
}
.logoImgStyle {
    max-width: 155px;
}

.forgetPassword {
    font-size: 16px !important;
    font-weight: bold;
    color: #3f3f3f;
    text-align: right;
}

@media only screen and (max-width: 900px) {
    #rightBoxContainer {
        height: 50vh;
        padding: 10px 20px;
        min-height: 400px;
    }
    #linkToSignUpText {
        width: 100%;
        font-size: 14px;
        margin-top: 30px;
        padding-bottom: 20px;
    }
    #backToHomeContainer {
        display: none;
    }
    .fullGrid {
        margin-top: 0 !important;
        width: 100% !important;
        overflow: auto;
    }
    .forgetPassword {
        font-size: 14px !important;
        margin-top: -5px !important;
    }
    .contentForm {
        padding-right: 0 !important;
    }
    .inputText div.MuiInput-root {
        height: 48px !important;
        margin: 0 !important;
    }
    .MuiGrid-spacing-xs-3 > .MuiGrid-item {
        padding-top: 0 !important;
    }
    .inputText img {
        top: 12px !important;
    }
    .logoContainerStyle { 
        margin-top: 0 !important;
        margin-bottom: 10px !important;
    }
    .logoImgStyle {
        max-width: 80px;
    }
    span.loginHeader {
        font-size: 22px !important;
    }
    button.loginHeader {
        font-size: 16px !important;
        margin-top: 5px !important;
    }
    a.loginHeader {
        font-size: 16px !important;
        margin-top: 5px !important;
    }
}
