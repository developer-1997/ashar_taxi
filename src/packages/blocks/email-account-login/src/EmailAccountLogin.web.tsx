import React from "react";

// Customizable Area Start
import {
  Box,
  Button,
  Input,
  Typography,
  InputAdornment,
  IconButton,
  InputLabel,
  Grid,
} from "@material-ui/core";
import { createTheme, styled, ThemeProvider } from "@material-ui/core/styles";
import VisibilityOff from "@material-ui/icons/VisibilityOff";
import Visibility from "@material-ui/icons/Visibility";
import { Formik } from "formik";
import {Link} from "react-router-dom"
import { lock, downArrow, logoImg } from './assets'
import EmailAccountLoginController, {
  Props,
} from "./EmailAccountLoginController";
import SharableLeftSidetaxi from "../../../components/src/SharableLeftSideTaxi.web";
import "./EmailAccountLogin.web.css";

export const LoginForm = ({
  values,
  handleChange,
  state,
  handleClickShowPassword,
  handleSubmit,
  handleFormValidation,
  errors,
  touched,
  handleStateError
}: any) => {

  const handleInputHandler = (e: any, errorType: any) => {
    handleChange(e)
    setTimeout(() => {
      handleStateError(errorType)
    }, 0)
  }

  return (
    <>
      <Grid container spacing={3} className="inputText fullGrid">
        <Grid item xs={12} md={10} style={email_login_styles.email_password_Style}>
          <InputLabel style={email_login_styles.formInputLabel} shrink htmlFor="bootstrap-input" className="fontWeightMedium">
            Email
          </InputLabel>

          <FormInputContainer style={email_login_styles.formContainerStyle}>
            <img style={email_login_styles.arrowImageStyle} src={downArrow} alt="" />
            <Input
              data-test-id="txt_Input_Email"
              type="text"
              name="attributes.email"
              placeholder={"Enter email"}
              fullWidth={true}
              value={values.attributes?.email}
              style={email_login_styles.formInputOutline}
              onChange={(e) => {
                handleInputHandler(e, "email")
              }}
            />
          </FormInputContainer>
          {handleFormValidation(errors?.attributes?.email, touched?.attributes?.email)}
          <div className="errorText"> {state.error?.email}</div>
        </Grid>
        <Grid item xs={12} md={10} style={email_login_styles.email_password_Style}>
          <InputLabel style={email_login_styles.formInputLabel} shrink htmlFor="bootstrap-input" className="fontWeightMedium">
            Password
          </InputLabel>

          <FormInputContainer style={email_login_styles.formContainerStyle}>
            <img style={email_login_styles.arrowImageStyle} src={lock} alt="" />
            <Input
              data-test-id="txt_Input_Password"
              type={state.enablePasswordField ? "password" : "text"}
              name="attributes.password"
              placeholder={"Enter password"}
              fullWidth={true}
              style={email_login_styles.formInputOutline}
              value={values.attributes?.password}
              onChange={(e) => {
                handleInputHandler(e, "password")
              }}
              endAdornment={
                <InputAdornment position="end">
                  <IconButton
                    aria-label="toggle password visibility"
                    onClick={handleClickShowPassword}
                    edge="end"
                  >
                    {state.enablePasswordField ? (<VisibilityOff />) : (<Visibility />)}
                  </IconButton>
                </InputAdornment>
              }
            />
          </FormInputContainer>
          {handleFormValidation(errors?.attributes?.password, touched?.attributes?.password)}
          <div className="errorText"> {state.error?.password}</div>
        </Grid>
        <Grid item xs={12} md={10} style={email_login_styles.forgetPasswordMainGrid}>
          <Link to="/ForgotPassword" style={{textDecoration: "none"}}>
            <Typography className="loginHeader forgetPassword">Forgot Password?</Typography>
          </Link>
        </Grid>
        <Grid item xs={12} md={10} style={email_login_styles.email_password_Style}>
          <Button
            data-test-id={"btnEmail_login"}
            variant="contained"
            color="primary"
            fullWidth
            className="loginHeader"
            style={email_login_styles.verifyBtn}
            onClick={() => {
              handleSubmit()
            }}
          >
            Continue{/*UI Engine::From Sketch*/}
          </Button>
        </Grid>
        <Grid item xs={12} style={email_login_styles.email_password_Style}>
          <Box display="flex" flexDirection="row" justifyContent="center">
            <Link
              data-test-id={"login_with_otp"}
              to="/EmailAccountLoginWithMobileNo"
              style={email_login_styles.linkContent}
              className="loginHeader">
              Login with OTP
            </Link>
          </Box>
        </Grid>
      </Grid>
    </>
  )
}
// Customizable Area End

const theme = createTheme({
  palette: {
    primary: {
      main: "#0000ff",
      contrastText: "#fff",
    },
  },
});

const LinkActive: any = styled('a')({
  fontSize: '26px'
});

export default class EmailAccountLogin extends EmailAccountLoginController {
  constructor(props: Props) {
    super(props);
  }
  handleFormValidation(error: any, touched: any) {
    return (
      <div className="errorText">
        {error && touched ? error : ""}
      </div>
    )
  }
  render() {
    return (
      // Required for all blocks
      <ThemeProvider theme={theme}>
        <Grid container>
          <Grid item md={12} lg={6}>
            <SharableLeftSidetaxi />
          </Grid>
          <Grid item md={12} lg={6}>
            <Box id="rightBoxContainer">
              <Box className="logoContainerStyle" display="flex" flexDirection="row" justifyContent="flex-end">
                <Link to="LandingPage">
                  <img src={logoImg} alt="Logo" className="logoImgStyle" />
                </Link>
              </Box>
              <Box className="contentForm">
                <Box style={email_login_styles.inHeading}>
                  <span data-test-id={"sign-in-route"} style={email_login_styles.isHeadValueActive} className="loginHeader">Sign In</span>
                </Box>
                <Formik
                  initialValues={{
                    type: "email_account",
                    attributes: {
                      email: "",
                      password: ""
                    }
                  }
                  }
                  validationSchema={this.emailLoginValidation}
                  onSubmit={(values) => {
                    this.createLoginForWeb(values)
                  }}
                >
                  {({
                    values,
                    errors,
                    touched,
                    handleChange,
                    handleSubmit,
                    setFieldValue,
                  }) => (
                    <LoginForm
                      data-test-id="registrationWrapperId"
                      values={values}
                      handleChange={handleChange}
                      errors={errors}
                      touched={touched}
                      state={this.state}
                      handleClickShowPassword={this.handleClickShowPassword}
                      setFieldValue={setFieldValue}
                      handleSubmit={handleSubmit}
                      setState={this.setState}
                      handleFormValidation={this.handleFormValidation}
                      routeHandler={this.routeHandler}
                      handleStateError={this.handleStateError}
                    />
                  )}
                </Formik>
              </Box>
              <Box flexGrow="1" width="100%" alignItems="flex-end" alignSelf="flex-end" display="flex" flexDirection="row" justifyContent="space-between">
                <Box id="linkToSignUpText" className="normalText" display="flex" flexDirection="row" justifyContent="center" alignItems="center">
                  Don't have an account?
                  <Link to="/EmailAccountRegistration" style={email_login_styles.linkContentNoUnderline}>
                    Sign Up
                  </Link>
                </Box>
                <Box id="backToHomeContainer">
                  <Link to="/LandingPage">
                    <Button style={email_login_styles.backbtn} className="normalText fontWeightMedium backToHomeButton"> Back to home</Button>
                  </Link>
                </Box>
              </Box>
            </Box>
          </Grid>
        </Grid>
      </ThemeProvider>
    );
  }
}

const FormInputContainer = styled('div')({

  ' & ::before': {
    display: 'none',
  },
  ' & ::after': {
    display: 'none',
  },

});
const email_login_styles: any = {
  // Customizable Area Start
  verifyBtn: {
    backgroundColor: '#090069',
    padding: '10px',
    borderRadius: '10px',
    textTransform: 'capitalize',
    fontSize: '20px',
    fontWeight: "bold",
    marginTop: "30px"
  },
  backbtn: {
    // position: "absolute",
    // bottom: "1.875rem",
    // right: "6.063rem",
    color: "#828282",
    fontWeight: "600",
    textDecoration: "underline",
    cursor: "pointer",
},
  formInputLabel: {
    color: '#242424',
    fontWeight: 'bold',
    fontSize: '18px',
  },
  formInputOutline: {
    border: '1px solid #c5c5c5',
    padding: '5px 15px 5px 60px',
    borderRadius: '10px',
    margin: '5px 0',
    fontWeight: '700',
    fontSize: '15px',
    width: '100%',
    height: "60px"
  },
  forgetPassword: {
    textAlign: 'right',
    fontWeight: 'bold',
    fontSize: '14px',
    color: "#242424"
  },
  email_password_Style: {
    margin: 'auto',
    minHeight: 'min-content'
  },
  sign_up_link_text: {
    margin: '0 auto 60px',
    textAlign: 'center'
  },
  formContainerStyle: {
    position: 'relative'
  },
  arrowImageStyle: {
    width: '25px', position: 'absolute', left: '18px', top: '22px'
  },
  errorStyle: {
    color: "red"
  },
  forgetPasswordMainGrid:
  {
    margin: 'auto',
    padding: '0 15px'
  },
  linkContent: {
    textDecoration: 'underline',
    color: '#090168',
    display: 'block',
    fontSize: '24px',
    cursor: "pointer"
  },
  linkContentNoUnderline: {
    textDecoration: 'none',
    color: '250068',
    fontWeight: 'bold',
    cursor: "pointer",
    marginLeft: '10px'
  },
  inHeading: {
    display: 'flex',
    justifyContent: 'center',
    width: '100%',
    fontSize: '32px',
    boxSizing: 'border-box',
    minHeight: 'min-content'
  },
  isHeadValue: {
    color: '#908f9d',
    textDecoration: 'none',
    position: 'relative',
    paddingBottom: '15px',
  },
  isHeadValueBox: {
    width: '50%',
    textAlign: 'center',
    fontSize: '26px',
    fontWeight: 'bold',
  },
  isHeadValueBoxRht: {
    width: '50%',
    textAlign: 'center',
    fontSize: '26px',
    fontWeight: 'bold',
    borderRight: '1px solid #ddd'
  },
  isHeadValueActive: {
    color: '#090069',
    textDecoration: 'none',
    paddingBottom: '0',
    margin: 0
  },
  container: {
    flex: 1,
    padding: 16,
    marginLeft: "auto",
    marginRight: "auto",
    maxWidth: 650,
    backgroundColor: "#fff",
    height: '100%'
  },
  // Customizable Area End
};