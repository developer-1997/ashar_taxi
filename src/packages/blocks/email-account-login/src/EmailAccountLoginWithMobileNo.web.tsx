import React from "react";

// Customizable Area Start
import {
    Box,
    Button,
    Grid
} from "@material-ui/core";
import { styled } from "@material-ui/core/styles";
import { Formik } from "formik";
import { logoImg } from './assets';
import { Link } from "react-router-dom"
import { LoginWithMobileNo } from "../../../components/src/LoginWithMobileNoForm.web";

import EmailAccountLoginController, {
    Props,
} from "./EmailAccountLoginController";
import SharableLeftSidetaxi from "../../../components/src/SharableLeftSideTaxi.web";
import "./loginWithMobileNo.css";

// Customizable Area End



export default class EmailAccountLoginWithMobileNo extends EmailAccountLoginController {
    constructor(props: Props) {
        super(props);
    }
    handleFormValidationForLogin(error: any, touched: any) {
        return (
            <div className="errorText">
                {error && touched ? error : ""}
            </div>
        )
    }
    render() {
        return (
            // Required for all blocks
            <Grid container>
                <Grid item md={12} lg={6} style={{width: "100%"}}>
                    <SharableLeftSidetaxi />
                </Grid>
                <Grid item md={12} lg={6} style={{ width: "100%", position: "relative" }}>
                    <Box id="rightBoxContainer">
                        <Box className="logoContainerStyle" display="flex" flexDirection="row" justifyContent="flex-end">
                            <Link to="/LandingPage">
                                <img src={logoImg} alt="Logo" className="logoImgStyle" />
                            </Link>
                        </Box>
                        <Box className="contentForm">
                            <Box
                                display="flex"
                                justifyContent="center"
                                alignItems="center"
                            >
                                <LinkActiveForOtpLogin varint="h4" component="h2" style={login_with_otp_ForWeb.isHeadValueActive} >
                                    <span className="loginHeader">Login with OTP</span>
                                </LinkActiveForOtpLogin>
                            </Box>
                            <Formik
                                initialValues={{
                                    type: "sms_account",
                                    attributes: {
                                        full_phone_number: ""
                                    }
                                }
                                }
                                validationSchema={this.emailLoginOtpValidation}
                                onSubmit={this.LoginAccountWithPhoneForWeb}
                            >
                                {({
                                    values,
                                    errors,
                                    touched,
                                    handleChange,
                                    handleSubmit,
                                    setFieldValue,
                                }) => (
                                    <LoginWithMobileNo
                                        data-test-id="registrationWrapperId"
                                        values={values}
                                        handleChange={handleChange}
                                        errors={errors}
                                        touched={touched}
                                        state={this.state}
                                        handleClickShowPassword={this.handleClickShowPassword}
                                        setFieldValue={setFieldValue}
                                        handleSubmit={handleSubmit}
                                        handleStateError={this.handleStateError}
                                        handleFormValidationForLogin={this.handleFormValidationForLogin}

                                    />
                                )}
                            </Formik>
                        </Box>
                        <Box
                            className="backToHomeContainer"
                            display="flex"
                            flexDirection="row"
                            justifyContent="flex-end"
                        >
                            <Link to="/LandingPage">
                                <Button className="normalText fontWeightMedium backToHomeButton">
                                    Back to home
                                </Button>
                            </Link>
                        </Box>
                    </Box>
                </Grid>
            </Grid>
        );
    }
}

const LinkActiveForOtpLogin: any = styled('a')({
    fontSize: '26px',
    textAlign: 'center'
    // '&::before': {
    //     content: '""',
    //     position: 'absolute',
    //     width: '20px',
    //     borderRadius: 5,
    //     bottom: 0,
    //     left: '50%',
    //     transform: 'translateX(-50%)',
    //     borderBottom: '5px solid #090069',
    // },
});


const login_with_otp_ForWeb: any = {
    // Customizable Area Start
    formInputOutline: {
        border: '1px solid #c5c5c5',
        padding: '5px 15px 5px 60px',
        borderRadius: '10px',
        margin: '5px 0',
        fontWeight: '700',
        fontSize: '15px',
        width: '100%',
        height: "60px"
    },
    inHeading: {
        display: 'flex',
        justifyContent: 'center',
        width: '100%',
        marginTop: "20px"
    },
    // errorMessage: {
    //     color: "red"
    // },
    mainContainer: { margin: '5rem auto' },
    isHeadValue: {
        color: '#908f9d',
        textDecoration: 'none',
        position: 'relative',
        paddingBottom: '15px',
    },
    verifyBtn: {
        backgroundColor: '#090069',
        padding: '15px',
        borderRadius: '10px',
        textTransform: 'capitalize',
        fontSize: '18px',
        fontWeight: "bold"
    },
    formInputLabel: {
        transform:"none",
        color: '#242424',
        fontWeight: 'bold',
        marginTop: '-20px',
        fontSize: '16px',
    },
    isHeadValueBox: {
        width: '50%',
        textAlign: 'center',
        fontSize: '26px',
        fontWeight: 'bold',
    },
    backbtn: {
        // position: "absolute",
        // bottom: "1.875rem",
        // right: "6.063rem",
        color: "#828282",
        fontWeight: "600",
        textDecoration: "underline",
        cursor: "pointer",
    },
    constainerBlock: { margin: 'auto' },
    isHeadValueActive: {
        display: 'inline',
        color: '#090069',
        textDecoration: 'none',
        position: 'relative',
        paddingBottom: '15px',
        margin: 0,
    },
    container: {
        flex: 1,
        padding: 16,
        marginLeft: "auto",
        marginRight: "auto",
        maxWidth: 650,
        backgroundColor: "#fff",
        height: '100%'
    },
    // Customizable Area End
};
