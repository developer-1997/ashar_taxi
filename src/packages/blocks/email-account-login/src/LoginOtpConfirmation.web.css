.logoContainerStyle { 
    margin-bottom: 40px;
    width: 100%;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
}
.logoImgStyle {
    max-width: 155px;
}
#rightBoxContainer {
    height: 100vh;
    margin: 0;
    padding: 30px;
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: flex-start;
    box-sizing: border-box;
}
.contentForm {
    padding-left: 60px;
    max-width: 500px;
    box-sizing: border-box;
}
.otpFormTitle {
    font-size: 26px !important;
    text-align: center;
    color: #090168;
    margin-bottom: 40px !important;
}
.otpFormParagraph {
    font-size: 20px;
    color: #8f8f9c;
    margin-bottom: 40px !important;
}
.otpFormParagraph b {
    display: block;
    color: #1f1f1f;
}
.isResendHeading {
    font-size: 20px !important;
    text-align: center;
    width: 100%;
    font-weight: 700;
    color: #3d3d3d;
    margin-top: 40px !important;
}
.backToHomeContainer {
    flex-grow: 1;
    width: 100%;
    align-items: flex-end;
}
.backbtn {
    color: #828282;
    font-weight: 600;
    text-decoration: underline !important;
    cursor: pointer;
    font-size: 18px !important;
}
@media only screen and (max-width: 900px) {
    #rightBoxContainer {
        height: 50vh;
        padding: 10px 20px;
        align-items: center;
        min-height: 400px;
    }
    .contentForm {
        padding-left: 0 !important;
        padding-right: 0 !important;
        width: auto !important;
    }
    .logoContainerStyle {  
        margin-bottom: 20px;
    }
    .logoImgStyle {
        max-width: 80px;
    }
    .otpFormTitle {
        font-size: 24px !important;
        margin-bottom: 20px !important;
    }
    .otpFormParagraph {
        font-size: 14px !important;
        margin-bottom: 20px !important;
    }
    .isResendHeading {
        font-size: 14px !important;
        margin-top: 5px !important;
    }
    .MuiInputBase-fullWidth {
        min-height: 60px !important;
        font-size: 15px !important;
    }
    .MuiButton-fullWidth {
        font-size: 16px !important;
        margin-top: 0 !important;
        padding: 10px !important;
    }
    .MuiGrid-spacing-xs-3 {
        width: 100% !important;
        margin: 0 !important;
    }
    .MuiGrid-spacing-xs-3 > .MuiGrid-item {
        padding-left: 0px !important;
        padding-right: 0px !important;
    }
}
