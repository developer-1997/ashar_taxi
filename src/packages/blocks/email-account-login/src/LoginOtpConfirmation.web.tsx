import React from "react";
// Customizable Area Start
import {
    Box,
    Button,
    Typography,
    Grid,
} from "@material-ui/core";
import { Link } from "react-router-dom"
import {
    ThemeProvider,
    createTheme,
} from "@material-ui/core/styles";
import { Formik } from "formik";
import EmailAccountLoginController, {
} from "./EmailAccountLoginController";
import SharableLeftSidetaxi from "../../../components/src/SharableLeftSideTaxi.web";
import { OtpInputField } from "../../../components/src/OtpInputField.web";
import { logoImg } from "../../email-account-registration/src/assets"
import "./LoginOtpConfirmation.web.css";

const theme = createTheme({
    palette: {
        primary: {
            main: "#0000ff",
            contrastText: "#fff",
        },
    },
});

export default class LoginOtpConfirmation extends EmailAccountLoginController {

    render() {
        
        return (
            // Required for all blocks
            <ThemeProvider theme={theme}>
                <Grid container className="inputText">
                    <Grid item xs={12} lg={6}>
                        <SharableLeftSidetaxi />
                    </Grid>
                    <Grid item xs={12} lg={6}>
                        <Box id="rightBoxContainer">          
                            <Box className="logoContainerStyle" display="flex" flexDirection="row" justifyContent="flex-end">
                                <Link to="/LandingPage">
                                    <img src={logoImg} alt="Logo" className="logoImgStyle"/>
                                </Link>
                            </Box>
                            <Box className="contentForm">
                                <Typography
                                variant="h2"
                                className="otpFormTitle loginHeader"
                                >
                                    Sign In
                                </Typography>
                                <Typography variant="body1" component="h2" className="otpFormParagraph">
                                    Enter 4 Digit Verification code sent to <br />
                                    <b>XXXXXXX260</b>
                                </Typography>
                                <Formik
                                    initialValues={{
                                        otp1: "",
                                        otp2: "",
                                        otp3: "",
                                        otp4: "",
                                    }}
                                    onSubmit={this.handleOtpFormSubmit}
                                >
                                    {({ values, handleChange, handleSubmit }) => (
                                        <OtpInputField
                                            state={this.state}
                                            values={values}
                                            handleChange={handleChange}
                                            handleSubmit={handleSubmit}
                                            handleStateError={this.handleStateError}
                                        />
                                    )}
                                </Formik>
                                <Grid item xs={12}>
                                    <Box style={sign_up_styles_for_login_confirm.IsResend}>
                                        <Typography variant="h4" className="isResendHeading fontWeightMedium">
                                            Didn't receive code ? <span onClick={() => [this.LoginAccountWithPhoneForWeb({ attributes: { full_phone_number: localStorage.getItem("phone_number") } }), this.otpSendAgain()]} style={{ cursor: "pointer", color: "#090168" }} data-test-id="otpsendButton">Resend code</span>
                                        </Typography>
                                    </Box>
                                </Grid>
                            </Box>
                            <Box className="backToHomeContainer" display="flex" flexDirection="row" justifyContent="flex-end">
                                <Link to="/LandingPage">
                                    <Button className="normalText fontWeightMedium backbtn" data-test-id="backId">Back to home</Button>
                                </Link>
                            </Box>
                        </Box>
                    </Grid>
                </Grid>
            </ThemeProvider>
        );
    }
}


const sign_up_styles_for_login_confirm: any = {
    // Customizable Area Start
    body: {
        minHeight: "100vh",
    },
    container: {
        flex: 1,
        padding: 16,
        marginLeft: "auto",
        marginRight: "auto",
        maxWidth: 650,
        backgroundColor: "#fff",
        height: "100%",
    },
    backbtn: {
        position: "absolute",
        bottom: "1.875rem",
        right: "6.063rem",
        color: "#828282",
        fontWeight: "600",
        textDecoration: "underline",
        cursor: "pointer",
    },
    errorMessageOtp: { color: "red" },

    rhtBox: {
        display: "flex",
        flexDirection: "column",
        alignItems: "center",
        padding: "30px 30px",
        width: "100%",
        margin: "0px auto",
        boxSizing: 'border-box',
        height: '100vh',
        overflow: 'auto',
    },
    inSubHeading: {
        display: "flex",
        width: "100%",
        marginBottom: "40px",
    },
    isEmail: {
        width: "100%",
        fontSize: "18px",
        color: "#3f3f3f",
        marginTop: "20px"
    },
    IsResend: {
        display: "flex",
        width: "100%",
        marginBottom: "40px",
    },
    isResendHeading: {
        width: "100%",
        fontSize: "14px",
        textAlign: "center",
        fontWeight: 700,
        color: "#3d3d3d",
        marginTop: "45px"
    }
};