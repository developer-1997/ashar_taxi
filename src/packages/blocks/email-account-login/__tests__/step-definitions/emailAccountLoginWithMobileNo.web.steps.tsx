import React from "react";
import { defineFeature, loadFeature } from "jest-cucumber"
import { shallow, ShallowWrapper} from 'enzyme'
import * as helpers from '../../../../framework/src/Helpers'
import EmailAccountLoginWithMobileNo from "../../src/EmailAccountLoginWithMobileNo.web";
import { LoginWithMobileNo } from "../../../../components/src/LoginWithMobileNoForm.web";

jest.mock("formik")

const screenProps = {
    navigation: {
        navigate: jest.fn(),
    },
    id: "mobile-account-login-scenario",
    mainProps:"",
}

const mobileLoginProp={
    values:{
        full_phone_number:"",
    },
    handleStateError:jest.fn(),
    handleChange:jest.fn(),
    handleSubmit:jest.fn(),
    handleFormValidationForLogin:jest.fn(),
    errors:"",
    touched:""
}

const feature = loadFeature('./__tests__/features/email-account-login-with-mobileNo-scenerio.web.feature');

defineFeature(feature, (test) => {

    beforeEach(() => {
        jest.resetModules()
        jest.doMock('react-native', () => ({ Platform: { OS: 'web' } }))
        jest.spyOn(helpers, 'getOS').mockImplementation(() => 'web');
    });

    test('Login with Mobile No', ({ given, when, then }) => {
        let mobileLoginWrapper: ShallowWrapper;
        let mobileWrapper: ShallowWrapper;
        let instance: EmailAccountLoginWithMobileNo;

        given('I am a User loading login with mobile no page', () => {
            mobileLoginWrapper = shallow(<EmailAccountLoginWithMobileNo history={undefined} {...screenProps} />)
            mobileWrapper = shallow(<LoginWithMobileNo {...mobileLoginProp} />)
            expect(mobileLoginWrapper).toBeTruthy()
            expect(mobileWrapper).toBeTruthy()
        });

        when('I navigate to the login with mobile no Screen', () => {
            instance = mobileLoginWrapper.instance() as EmailAccountLoginWithMobileNo
        });

        then('Login with mobile no page will load with out error', () => {
            expect(mobileLoginWrapper).toBeTruthy();
            expect(mobileLoginWrapper).toMatchSnapshot();
        });

        then("I can handle the mobile number in the login", () => {
            let mobileInput = mobileWrapper.findWhere((node) => node.prop('data-test-id') === 'full_phone_number');
            mobileInput.simulate('change', 'mobileNo');
            instance.handleStateError("");
        })

        then("I can handle the submit mobile number in the login",()=>{
            let submitMobileNo = mobileWrapper.findWhere((node) => node.prop('data-test-id') === 'submit_mobile_no');
            submitMobileNo.simulate('click');
        })

        then("I can handle the form validation in the login with the mobile no",()=>{
            instance.handleFormValidationForLogin("","");
        })

        then("I can handle the user login by the mobile no",()=>{
            instance.userLoginByMobileNo("","");
        })

        then("I can handle the login account with phone number",()=>{
            instance.LoginAccountWithPhoneForWeb("");
        })

        then('I can leave the screen with out errors', () => {
            instance.componentWillUnmount();
            expect(mobileLoginWrapper).toBeTruthy();
        });
    });
})
