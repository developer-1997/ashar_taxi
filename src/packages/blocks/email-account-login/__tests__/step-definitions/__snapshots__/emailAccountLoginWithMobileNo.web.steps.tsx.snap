// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginWithMobileNo Login with Mobile No 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    lg={6}
    md={12}
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <SharableLeftSideTaxi />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    lg={6}
    md={12}
    style={
      Object {
        "position": "relative",
        "width": "100%",
      }
    }
  >
    <Styled(MuiBox)
      sx={
        Object {
          "alignItems": "center",
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "margin": "0px auto",
          "maxHeight": "100vh",
          "overflow": "hidden auto",
          "padding": "30px 30px",
          "width": "100%",
        }
      }
    >
      <Styled(MuiBox)
        className="logoContainerStyle"
        display="flex"
        flexDirection="row"
        justifyContent="flex-end"
      >
        <Link
          replace={false}
          to="/LandingPage"
        >
          <img
            alt="Logo"
            className="logoImgStyle"
            src={
              Object {
                "testUri": "../../../packages/blocks/email-account-login/assets/logo.png",
              }
            }
          />
        </Link>
      </Styled(MuiBox)>
      <Styled(MuiBox)
        alignItems="center"
        display="flex"
        justifyContent="center"
      >
        <Styled(a)
          component="h2"
          style={
            Object {
              "color": "#090069",
              "display": "inline",
              "margin": 0,
              "paddingBottom": "15px",
              "position": "relative",
              "textDecoration": "none",
            }
          }
          varint="h4"
        >
          <span
            className="loginHeader"
          >
            Login with OTP
          </span>
        </Styled(a)>
      </Styled(MuiBox)>
      <Formik
        initialValues={
          Object {
            "attributes": Object {
              "full_phone_number": "",
            },
            "type": "sms_account",
          }
        }
        onSubmit={[Function]}
        validationSchema={
          ObjectSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_defaultDefault": [Function],
            "_deps": Array [],
            "_excludedEdges": Array [],
            "_exclusive": Object {},
            "_mutate": undefined,
            "_nodes": Array [
              "attributes",
              "type",
            ],
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "object",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "fields": Object {
              "attributes": ObjectSchema {
                "_blacklist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "_conditions": Array [],
                "_defaultDefault": [Function],
                "_deps": Array [],
                "_excludedEdges": Array [],
                "_exclusive": Object {},
                "_mutate": undefined,
                "_nodes": Array [
                  "full_phone_number",
                ],
                "_options": Object {
                  "abortEarly": true,
                  "recursive": true,
                },
                "_type": "object",
                "_typeError": [Function],
                "_whitelist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "fields": Object {
                  "full_phone_number": StringSchema {
                    "_blacklist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "_conditions": Array [],
                    "_deps": Array [],
                    "_exclusive": Object {
                      "required": true,
                    },
                    "_mutate": undefined,
                    "_options": Object {
                      "abortEarly": true,
                      "recursive": true,
                    },
                    "_type": "string",
                    "_typeError": [Function],
                    "_whitelist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "tests": Array [
                      [Function],
                    ],
                    "transforms": Array [
                      [Function],
                    ],
                    "type": "string",
                  },
                },
                "tests": Array [],
                "transforms": Array [
                  [Function],
                ],
                "type": "object",
              },
              "type": StringSchema {
                "_blacklist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "_conditions": Array [],
                "_deps": Array [],
                "_exclusive": Object {
                  "required": true,
                },
                "_mutate": undefined,
                "_options": Object {
                  "abortEarly": true,
                  "recursive": true,
                },
                "_type": "string",
                "_typeError": [Function],
                "_whitelist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "tests": Array [
                  [Function],
                ],
                "transforms": Array [
                  [Function],
                ],
                "type": "string",
              },
            },
            "tests": Array [],
            "transforms": Array [
              [Function],
            ],
            "type": "object",
          }
        }
      >
        <Component />
      </Formik>
    </Styled(MuiBox)>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    >
      <Styled(MuiBox)>
        <Link
          replace={false}
          to="/LandingPage"
        >
          <WithStyles(ForwardRef(Button))
            className="normalText fontWeightMedium"
            style={
              Object {
                "bottom": "1.875rem",
                "color": "#828282",
                "cursor": "pointer",
                "fontWeight": "600",
                "position": "absolute",
                "right": "6.063rem",
                "textDecoration": "underline",
              }
            }
          >
             Back to home
          </WithStyles(ForwardRef(Button))>
        </Link>
      </Styled(MuiBox)>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
